Step2 - 엔티티 매핑

limwoobin opened the pull request:
안녕하세요 재연님
2 단계 리뷰 요청드립니다! 
잘부탁드립니다 ㅎㅎ

limwoobin committed:
docs: Step2 기능 요구사항 작성

limwoobin committed:
feat: DeleteHistory 에 대해 deleteById 를 객체인 User 로 매핑하라

- 매핑 변경에 따른 테스트 코드를 작성하라

limwoobin committed:
feat: Answer 의 연관관계를 객체로 변경하라

- 이에 맞게 answerRepository 의 query method 네이밍을 수정하라

limwoobin committed:
test: Answer 의 연관관계를 테스트 코드를 작성해 검증하라

limwoobin committed:
fix: 테스트 코드 id 값 비교로직에서 변수를 변경하라

limwoobin committed:
feat: Question 의 writer 를 User 객체로 변경하라

- 변경에 따라 테스트 코드를 작성해 검증한다

limwoobin committed:
feat: 연관관계가 객체로 변경됨에 따라 QnaService 로직을 객체 연관관계로 변경하라

jaeyeonling commented on src/main/java/qna/domain/Answer.java:
```diff
@@ -47,29 +54,29 @@ public Answer(Long id, User writer, Question question, String contents) {
 
     public boolean isOwner(User writer) {
-        return this.writerId.equals(writer.getId());
+        return this.writer.getId().equals(writer.getId());
```
get-을 사용하지 않고 메시지를 던져보면 어떨까요?

limwoobin replied:
넵 그렇게되면 책임이 명확해지겠군여
감사합니다 ㅎㅎ

jaeyeonling commented on src/test/java/qna/repository/AnswerRepositoryTest.java:
```diff
@@ -4,90 +4,157 @@
+            List<Answer> answers = answerRepository.findAll();
+            answerRepository.delete(answer);
+
+            List<Answer> deleted_answers = answerRepository.findAll();
```
변수명을 스네이크 케이스로 작성해주신 이유가 있을까요?

limwoobin replied:
앗 실수네요 카멜케이스로 변경하겠습니다 ㅎㅎ

jaeyeonling commented on src/main/java/qna/service/QnaService.java:
```diff
@@ -39,7 +39,7 @@ public void deleteQuestion(User loginUser, Long questionId) throws CannotDeleteE
             throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
         }
 
```
지금은 삭제에 대한 역할을 서비스에서 담당하고 있는데요.
`deleteQuestion` 메서드 내부 구현을 `Question` 내부로 옮겨볼 수 있을 것 같습니다 :)

limwoobin replied:
QnaService 쪽 로직은 3단계 리팩토링때 진행하려고 안건드렸는데
이 부분은 3단계때 옮겨봐도 될까요??

jaeyeonling replied:
앗 제가 착각했네요 ㅎㅎ 
해당 내용은 3단계에서 진행해주세요 👍 👍

jaeyeonling reviewed (changes requested):
2단계 미션을 잘 구현해주셨습니다 👍 
코멘트 남겨두었으니 확인 부탁드릴게요!!

limwoobin committed:
refactor: pr review 를 적용하라

- 필드의 get를 사용하지 말고 메시지를 던져서 처리하라
- 테스트 코드의 변수명을 snake case 에서 camel case 로 변경하라

jaeyeonling commented on src/main/java/qna/domain/Answer.java:
```diff
@@ -47,29 +54,30 @@ public Answer(Long id, User writer, Question question, String contents) {
 
     public boolean isOwner(User writer) {
-        return this.writerId.equals(writer.getId());
+//        return this.writer.getId().equals(writer.getId());
```
불필요한 주석은 제거해주세요~

limwoobin replied:
넵 제가 놓쳤네요ㅠ
리뷰 감사합니다~

jaeyeonling reviewed (approved):
코멘트를 잘 반영해주셨습니다 💯 
바로 다음 단계 진행해주세요 😄
